src/routes.jsx

```diff
@@ -0,0 +1,319 @@
+                const {
+                  default: ExternalCourseEnrollment,
+                  makeExternalCourseEnrollmentLoader,
+                } = await import('./components/course/routes/ExternalCourseEnrollment');
+                return {
+                  Component: ExternalCourseEnrollment,
+                  loader: getRouteLoader(makeExternalCourseEnrollmentLoader, queryClient),
+                };
+              },
+            },
+            {
+              path: 'complete',
+              lazy: async () => {
+                const { default: ExternalCourseEnrollmentConfirmation } = await import('./components/course/routes/ExternalCourseEnrollmentConfirmation');
+                return {
+                  Component: ExternalCourseEnrollmentConfirmation,
+                };
+              },
+            },
+          ],
+        },
+      ],
+    },
+    {
+      path: 'licenses/:activationKey/activate',
+      lazy: async () => {
+        const { default: LicenseActivationRoute } = await import('./components/app/routes/LicenseActivationRoute');
+        return {
+          Component: LicenseActivationRoute,
+        };
+      },
+      errorElement: (
+        <RouteErrorBoundary
+          showSiteHeader={false}
+          showSiteFooter={false}
+        />
+      ),
+    },
+    {
+      path: 'videos/:videoUUID',
+      lazy: async () => {
+        const { makeVideosLoader, VideoDetailPage } = await import('./components/microlearning');
+        return {
+          Component: VideoDetailPage,
+          loader: makeVideosLoader(queryClient),
+        };
+      },
+      errorElement: (
+        <RouteErrorBoundary
+          showSiteHeader={false}
+          showSiteFooter={false}
+        />
+      ),
+    },
+  ];
+  if (hasNotFoundRoutes) {
+    enterpriseSlugChildRoutes.push({
+      path: '*',
+      element: <NotFoundPage />,
+    });
+  }
+  const enterpriseSlugRoutes = [
+    {
+      path: ':enterpriseSlug?',
+      loader: getRouteLoader(makeRootLoader, queryClient),
+      element: <Layout />,
+      children: enterpriseSlugChildRoutes,
+    },
+  ];
+  return enterpriseSlugRoutes;
+}
+
+function getOtherRoutes() {
+  const otherRoutes = [
+    {
+      path: 'invite/:enterpriseCustomerInviteKey',
+      lazy: async () => {
+        const {
+          default: EnterpriseInviteRoute,
+          makeEnterpriseInviteLoader,
+        } = await import('./components/app/routes/EnterpriseInviteRoute');
+        return {
+          Component: EnterpriseInviteRoute,
+          loader: makeEnterpriseInviteLoader(),
+        };
+      },
+    },
+  ];
+  return otherRoutes;
+}
+
+export function getRoutes({ queryClient, options = {} }) {
+  const {
+    hasNotFoundRoutes = true,
+  } = options;
+  const enterpriseSlugRoutes = getEnterpriseSlugRoutes({ queryClient, hasNotFoundRoutes });
+  const rootChildRoutes = [
+    ...getOtherRoutes(),
+    ...enterpriseSlugRoutes,
+  ];
+  if (hasNotFoundRoutes) {
```

Comment:
[inform] need to sanity check if the two conditional `NotFoundPage` routes (based on `hasNotFoundRoutes`) are still needed now that the approach has changed slightly.

Hunch is these two routes can safely always be included now, simplifying this logic slightly.